tokio-trace-futures/examples/proxy_server.rs

```diff
@@ -0,0 +1,144 @@
+/*
+This example has been taken and modified from here :
+https://raw.githubusercontent.com/tokio-rs/tokio/master/tokio/examples/proxy.rs
+*/
+
+extern crate futures;
+extern crate tokio;
+#[macro_use]
+extern crate tokio_trace;
+extern crate tokio_trace_fmt;
+extern crate tokio_trace_futures;
+
+use tokio_trace::field;
+use tokio_trace_futures::Instrument;
+
+use std::env;
+use std::io::{self, Read, Write};
+use std::net::{Shutdown, SocketAddr};
+use std::sync::{Arc, Mutex};
+
+use tokio::io::{copy, shutdown};
+use tokio::net::{TcpListener, TcpStream};
+use tokio::prelude::*;
+
+fn main() -> Result<(), Box<std::error::Error>> {
+    let listen_addr = env::args().nth(1).unwrap_or("127.0.0.1:8081".to_string());
+    let listen_addr = listen_addr.parse::<SocketAddr>()?;
+
+    let server_addr = env::args().nth(2).unwrap_or("127.0.0.1:3000".to_string());
+    let server_addr = server_addr.parse::<SocketAddr>()?;
+
+    // Create a TCP listener which will listen for incoming connections.
+    let socket = TcpListener::bind(&listen_addr)?;
+    println!("Listening on: {}", listen_addr);
+    println!("Proxying to: {}", server_addr);
+
+    let done = socket
+        .incoming()
+        .map_err(|e| debug!(msg = "error accepting socket", error = field::display(e)))
+        .for_each(move |client| {
+            let server = TcpStream::connect(&server_addr);
+            match client.peer_addr() {
+                Ok(x) => info!(
```

Comment:
> Ah, that's because no events are actually being recorded inside that span. If this wasn't the case (i.e. if we had enabled events emitted by `tokio` or another dependency, or if we added events in that span) we would see it in the logs as well.

How about this ?

Reply:
I mean the Ok(x) => info!(

Reply:
shouldn't that be printed? also another thing to notice is that the outer most scope prints `proxy_server` despite being instrumented as `proxy`.

Reply:
@kanishkarj Ah, I see the problem here --- the "proxy" span is being created outside of the subscriber context, so it's not being recorded. 

If you moved the call to `instrument` on the `for_each` future to inside the `subscriber::with_default` block, like so: 
```rust
    tokio_trace::subscriber::with_default(subscriber, || {
        let done = done.instrument(span!("proxy", listen_addr = field::debug(&listen_addr)));
        tokio::run(done);
    });
```

you'll see everything is now inside the "proxy" span.

Reply:
> shouldn't that be printed? also another thing to notice is that the outer most scope prints `proxy_server` despite being instrumented as `proxy`.

Ah, what you're seeing here is the _target_ that all these events were recorded with, not the outermost span --- if you look at all the events that this example logs, you'll see that they say `proxy_server:` _after_ the innermost span's name. 

Targets are separate concept from spans --- they refer to the _part_ of the system that an event or span relates to. By default, the target of any `tokio-trace` span or event is the name of the Rust module where it's defined, but you can specify an alternate target as well.

Reply:
Ah got it.